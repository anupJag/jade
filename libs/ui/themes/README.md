# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
