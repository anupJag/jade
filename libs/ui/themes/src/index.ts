export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
