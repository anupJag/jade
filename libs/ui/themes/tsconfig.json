{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
