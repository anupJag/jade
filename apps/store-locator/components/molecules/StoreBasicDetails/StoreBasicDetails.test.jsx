import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';

import { StoreBasicDetails } from './index';

describe('StoreBasicDetails', () => {
  let StoreBasicDetailsComponent = '';

  beforeEach(() => {
    const basicDetails = [
      {
        icon: 'info',
        value: '',
      },
      {
        icon: 'phone',
        value: '',
      },
    ];

    StoreBasicDetailsComponent = render(
      <ThemeProvider>
        <StoreBasicDetails basicDetails={basicDetails} />
      </ThemeProvider>,
    );
  });

  it('should render store basic details correctly', async () => {
    const { asFragment } = StoreBasicDetailsComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
