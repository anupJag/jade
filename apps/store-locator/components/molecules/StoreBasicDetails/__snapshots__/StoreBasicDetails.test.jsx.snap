// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreBasicDetails should render store basic details correctly 1`] = `
<DocumentFragment>
  <div
    class="css-nmceeh"
  >
    <div
      class="css-1bs4ebi"
    >
      <div
        class="css-0"
      >
        <div
          class="css-1l21umq"
        >
          <svg
            aria-hidden="true"
            class="css-1im46kq"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="css-0"
      />
    </div>
    <div
      class="css-1bs4ebi"
    >
      <div
        class="css-0"
      >
        <div
          class="css-1l21umq"
        >
          <svg
            aria-hidden="true"
            class="css-1im46kq"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <path
              d="M2.20731,0.0127209 C2.1105,-0.0066419 1.99432,-0.00664663 1.91687,0.032079 C0.871279,0.438698 0.212942,1.92964 0.0580392,2.95587 C-0.426031,6.28627 2.20731,9.17133 4.62766,11.0689 C6.77694,12.7534 10.9012,15.5223 13.3409,12.8503 C13.6507,12.5211 14.0186,12.037 13.9993,11.553 C13.9412,10.7397 13.186,10.1588 12.6051,9.71349 C12.1598,9.38432 11.2304,8.47427 10.6495,8.49363 C10.1267,8.51299 9.79754,9.05515 9.46837,9.38432 L8.88748,9.96521 C8.79067,10.062 7.55145,9.24878 7.41591,9.15197 C6.91248,8.8228 6.4284,8.45491 6.00242,8.04829 C5.57644,7.64167 5.18919,7.19632 4.86002,6.73161 C4.7632,6.59607 3.96933,5.41495 4.04678,5.31813 C4.04678,5.31813 4.72448,4.58234 4.91811,4.2919 C5.32473,3.67229 5.63453,3.18822 5.16982,2.45243 C4.99556,2.18135 4.78257,1.96836 4.55021,1.73601 C4.14359,1.34875 3.73698,0.942131 3.27227,0.612963 C3.02055,0.419335 2.59457,0.0708094 2.20731,0.0127209 Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="css-0"
      />
    </div>
  </div>
</DocumentFragment>
`;
